app/models/status_message.rb

```diff
@@ -120,6 +89,29 @@ def post_location
     }
   end
 
+  def people_allowed_to_be_mentioned
+    @aspects_ppl ||=
+      if public?
+        "all"
+      else
+        [
+          *AspectMembership.where(aspect_id: aspects).map(&:person),
+          *share_visibilities.map {|sv| sv.user.person }
```

Comment:
This line is not needed, because `people_allowed_to_be_mentioned` is only used with the author of the post, and the people from the aspects (see above) are all you need here.